src/onefetch/language.rs

```diff
@@ -59,6 +73,66 @@ macro_rules! define_languages {
             vec![ $( tokei::LanguageType::$name ,)* ]
         }
 
+        #[cfg(test)]
+        mod true_colors {
+
+            use std::fmt;
+            use super::*;
+            use paste::paste;
+
+            impl fmt::Display for Colors {
+                fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                    let mut output = String::new();
+                    output += "Colors {\n";
+                    output += "    basic_colors: vec![\n";
+                    let mut color_index = 0;
+                    for bc in self.basic_colors.iter() {
+                        output += &format!( "        Color::{:?}, // {}\n", bc, color_index );
+                        color_index = color_index + 1;
+                    }
+                    output += "    ], \n";
+                    if let Some(tcs) = &self.true_colors {
+                        output += "    true_colors: vec![\n";
+                        let mut color_index = 0;
+                        for tc in tcs.iter() {
+                            output += &format!( "        Color::{:?}, // {}\n", tc, color_index );
+                            color_index = color_index + 1;
+                        }
+                    } else {
+                        output += "    true_colors: None\n";
+                    };
+                    output += "    ], \n";
+                    output += "}\n";
+                    write!( f, "{}", output )
+                }
+            }
+
+            $(
+                paste! {
+                    #[test]
+                    #[ignore]
+                    fn [<$name:lower _basic_color_test>] () {
+                        let colors = $colors;
+                        let mut color_index = 0;
+                        for bc in colors.basic_colors.iter() {
+                            let color_str = &format!( "Color::{:?}", bc );
+                            assert!( !color_str.contains( "TrueColor" ), " language {} has true color {} in basic colors at index {} please change to a basic color: Black, Red, Green, Yellow, Blue, Magenta, Cyan, White, BrightBlack, BrightRed, BrightGreen, BrightYellow, BrightBlue, BrightMagenta, BrightCyan, BrightWhite", $display, color_str, color_index );
+                            color_index = color_index + 1;
```

Comment:
```suggestion
                        for (index, bc) in colors.basic_colors.iter().enumerate() {
                            let color_str = &format!( "Color::{:?}", bc );
                            assert!( !color_str.contains( "TrueColor" ), " language {} has true color {} in basic colors at index {} please change to a basic color: Black, Red, Green, Yellow, Blue, Magenta, Cyan, White, BrightBlack, BrightRed, BrightGreen, BrightYellow, BrightBlue, BrightMagenta, BrightCyan, BrightWhite", $display, color_str, color_index );
```
Just a bit more readable IMO, and prevents you from having to declare a value outside the scope of the loop.

Reply:
Agree, see 36a9e9c